src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java

```diff
@@ -41,6 +41,21 @@ public GoDiscoveryContext(Service service, ApiaryConfig apiaryConfig) {
           .put(Field.Kind.TYPE_DOUBLE, "0.0")
           .build();
 
+  @Override
+  protected String arrayTypeName(String elementName) {
```

Comment:
In the java example, these methods have the syntax of type declarations. Here in Go, they don't. Are you saying in your comment that for Go we only use the output of these functions for the variable names? If so, it seems like you're overloading the use of the functions. It may be clearer to have arrayType() vs arrayName(). Or maybe I'm misunderstanding this.